controllers/resources/lease.go

```diff
@@ -108,8 +108,12 @@ func (m *nhcLeaseManager) ManageLease(ctx context.Context, remediationCR *unstru
 		m.log.Error(err, "managing lease - couldn't fetch lease", "node name", remediationCR.GetName())
 		return 0, err
 	}
+	//nothing to do with this lease
+	if !m.isLeaseOwner(nodeLease) {
+		return 0, nil
```

Comment:
We are not owner, but we don't know if the lease is expired. So I think we can try to take the lease IMHO.
Same goes for the isNotFound case above, just try to create a new one, as mentioned in an earlier comment.

example: https://github.com/slintes/node-healthcheck-operator/commit/e9f88ef6f5d820a5003815fb90125368849787cd#diff-b0df20bbf7681fe87317f506104c06107e4979f97a9e3698289a010890602672R85

Reply:
`ManageLease` only extends or removes expired leases (i.e it manages existing leases).
In case we need to take a lease (whether we are the owner or not) it will happen in `ObtainLease`

Reply:
`ManageLease` is handling leases for existing remediation CRs. IMHO this doesn't exclude creating a new lease if needed.

Reply:
IIUC we can abandon this thread, since we are already discussing the term on which a lease will be created in the [thread above](https://github.com/medik8s/node-healthcheck-operator/pull/212#discussion_r1267886016).